dockstore-yaml-proposals/jbrowse/.dockstore.yml

```diff
@@ -0,0 +1,65 @@
+# .dockstore.yml for JBrowse
+version: 1.1
+service:
+  type: docker-compose
+  name: JBrowse
+  author:
+  description: o
+    JBrowse is a fast, scalable genome browser built completely with JavaScript and HTML5.
+    It can run on your desktop, or be embedded in your website.
+  # These are files the Dockstore will index. They will be directly downloadable from Dockstore
+  files:
+     - docker-compose.yml
+     - healthcheck.sh
+     - port.sh
+     - prepare_refdata.sh
+     - stand_up.sh
+     - stop.sh
+     - GRCh37.json
+
+  scripts:
+    # The scripts preprovision, prestart, start, poststart, and postprovision will be executed in order.
+    # The only script required to have a value is start.
+
+    # Before any provisioning
+    preprovision:
+
+    # After provisioning, before starting the service
+    prestart: prepare_refdata.sh
+
+    # Starts the service
+    start: stand_up.sh
+
+    # After the service has started
+    poststart:
+
+    # After the service has started. For multiple provisionings, e.g., the user may decide to add more data later.
+    postprovision:
+
+    # Returns the port the service is exposing. Can be used by a client to know what port to proxy
+    port: port.sh
+
+    # Returns an exit code of 0 if service is running
+    healthcheck: healthcheck.sh
+
+    # Stops the service
+    stop: stop.sh
+
+  environment:
+    JBROWSE_NAME:
+        default: jbrowse
+        description: The name of the container.
+    HTTP_PORT:
+        default: "3000"
```

Comment:
Still kind of confused (think I asked this question before). 
Why is there a `port.sh` above that asks for the port if the port is specified as a parameter anyway? 
Shouldn't the user setting the `HTTP_PORT` parameter already know what the port is?

Reply:
The idea for port.sh is that a client, e.g., Leo, can call it to know what port is exposed, so that Leo can in turn reverse proxy it. 

In this case, the port is configurable, but there could be cases where it is not, and Leo would still need to know which port is exposed.

Reply:
Sounds good, that description should probably be in the schema version from above or similar

Reply:
Yeah, I think this will be a welcomed bit of conceptual glue. The problem becomes services may choose to expose a variety of ports so we may want to specify the response as a set of tuples of ports and names. We could specify the response as an array of named ports.

Reply:
I've thought about that and not sure how that would work with reverse proxying. If the reverse proxy terminates the cert, and also needs to route to multiple ports, that could get pretty complicated, as the routing would probably be based on paths.

I'm inclined to keep this simple for now, with just one port, and then when we run across a service that has multiple ports, cross that bridge then.